backend/api/contacts/src/test/java/co/airy/core/contacts/ImportContactsTest.java

```diff
@@ -0,0 +1,151 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.util.List;
+import java.util.Arrays;
+
+import static co.airy.test.Timing.retryOnException;
+import static org.apache.kafka.streams.KafkaStreams.State.RUNNING;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT, classes = AirySpringBootApplication.class)
+@TestPropertySource(value = "classpath:test.properties")
+@AutoConfigureMockMvc
+@ExtendWith(SpringExtension.class)
+public class ImportContactsTest {
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private WebTestHelper webTestHelper;
+
+    @Autowired
+    private Stores stores;
+
+    @RegisterExtension
+    public static final SharedKafkaTestResource sharedKafkaTestResource = new SharedKafkaTestResource();
+
+    private static KafkaTestHelper kafkaTestHelper;
+
+    @BeforeAll
+    static void beforeAll() throws Exception {
+        kafkaTestHelper = new KafkaTestHelper(sharedKafkaTestResource, Topics.getTopics());
+        kafkaTestHelper.beforeAll();
+    }
+
+    @AfterAll
+    static void afterAll() throws Exception {
+        kafkaTestHelper.afterAll();
+    }
+
+    @BeforeEach
+    void beforeEach() throws Exception {
+        webTestHelper.waitUntilHealthy();
+    }
+
+    @Test
+    void canImportContacts() throws Exception {
+        final List<CreateContactPayload> payload = mockContactsListPayload();
+
+        final String content = webTestHelper.post(
+                "/contacts.import",
+                objectMapper.writeValueAsString(payload))
+            .andExpect(status().isCreated()).andReturn().getResponse().getContentAsString();
+
+        List<ContactResponsePayload> contactsResp = objectMapper.readValue(
+                content,
+                new TypeReference<List<ContactResponsePayload>>() {});
+        contactsResp.stream().forEach((contactResponse) -> {
+            try {
+                retryOnException(() -> {
+                    final Contact c = stores.getContact(contactResponse.getId());
```

Comment:
In the spirit of testing the real thing I think we should test this using either the `/contacts.list` endpoint or by requesting each contact with `/contacts.info`